src/map/skill.cpp

```diff
@@ -3494,7 +3494,7 @@ int64 skill_attack (int attack_type, struct block_list* src, struct block_list *
 		}
 	}
 
-	if( dmg.flag&BF_MAGIC && ( skill_id != NPC_EARTHQUAKE || (battle_config.eq_single_target_reflectable && (flag&0xFFF) == 1) ) )
+	if(((tsc && tsc->data[SC_MAGICMIRROR] && skill_get_type(skill_id)==BF_MAGIC) || dmg.flag&BF_MAGIC) && ( skill_id != NPC_EARTHQUAKE || (battle_config.eq_single_target_reflectable && (flag&0xFFF) == 1) ) )
```

Comment:
This might just be me being picky but can we make this if multiple lines to something like the following as its getting hard to read
```
if(
   (
    (tsc && tsc->data[SC_MAGICMIRROR] && skill_get_type(skill_id)==BF_MAGIC) 
    || dmg.flag&BF_MAGIC
   ) && ( 
     skill_id != NPC_EARTHQUAKE || 
     ( battle_config.eq_single_target_reflectable && (flag&0xFFF) == 1 ) 
  ) 
)
```

that or define the